✨ Add section colors and section progress support

MrCartoon opened the pull request:
- Added support of automatic color recognition (with respect to AbleSet web colors) that can be applied to background and/or to text color.
- Based on previously implemented section and song progress added support for progress based on section number
- Added new "Queued' image (not perfect, but works fine)
- Increased section present to 48 (I just have some song that have more sections, I'm fine rejecting this update in particular)

https://github.com/user-attachments/assets/b58f33f6-9e64-495e-8c9d-d24acec46399

leolabs commented:
Thank you so much for this PR, that's impressive work! 😍 

The code looks good to me, I'll try this out on my Stream Deck soon to ensure everything works as expected.

Regarding the "Queued" icon, I could imagine a Play icon or an arrow to the right would work to indicate that a song is queued up. Alternatively, maybe a big "Q" could also work, though it might not be as pretty as an icon. What do you think?

I'm looking forward to your reply!

MrCartoon commented:
Thanks for quick reply

I'm not sure that I understand speaking of icon, I didn't touch anything about songs except adding transparent progress as to any else.
Why I added Queued icon is because the default configuration just sets background to green when section is queued and having green being used as regular section color makes me lose what section I queued.

I guess I'm all good with your approach so long the queued section/song is easily noticable.

Also just realised that I've added support only to absolute sections, but not to relative. Might do it later.

leolabs commented:
My bad, I meant this in regard to sections, not songs. Though it might be nice to see which song is queued up next as well. This might be something for a future update 😊 

> Also just realised that I've added support only to absolute sections, but not to relative. Might do it later.

Adding support for relative sections would be great, especially for users with smaller Stream Decks!

Let me know if you have any further questions or feedback!

MrCartoon commented:
Here's relative update.

Also figured out I don't need to expose all variables to companion like sectoinsCount and section${i}Color.

Speaking of queued icon, it's up to you what to do.

https://github.com/user-attachments/assets/4c13e5e2-956f-46fe-b9b3-8545644137d7

leolabs commented:
Perfect, thank you so much! I'll merge this and will release a new version of the Companion module soon 😊
